text/metrics/0000-pipeline-monitoring.md

```diff
@@ -0,0 +1,347 @@
+ratio requires all three timeseries.  If the processor is a sampler,
+for example, the effective sampling rate is computed as
+`(accepted+refused)/(accepted+refused+dropped)`.
+
+While the collector defines and emits metrics sufficient for
+monitoring the individual pipeline component--taken as a whole, there
+is substantial redundancy in having so many exclusive counters.  For
+example, when a collector pipeline features no processors, the
+receiver's `refused` count is expected to equal the exporter's
+`send_failed` count.
+
+When there are several processors, it is primarily the number of
+dropped items that we are interested in counting.  Whene there are
+multiple sequential processors in a pipeline, however, counting the
+total number of items at each stage in a multi-processor pipeline
+leads to over-counting in aggregate.  For example, if you combine
+`accepted` and `refused` for two adjacent processors, then remove the
+metric attribute which distinguishes them, the resulting sum will be
+twice the number of items processed by the pipeline.
+
+The same logic applies to suggest that multiple sequential collectors
+in a pipeline cannot use the same metric names, otherwise removal of
+the which distinguishing metric attribute would cause over-counting of
+the pipeline.
+
+### Pipeline monitoring
+
+The term _Stage_ is used to describe the a single component in an
+export pipeline.
+
+The term _Station_ is used to describe a location in the export
+pipeline where the participating stages are part of the same logical
+failure domain.  Typically each SDK or Collector is considered a
+station.
+
+#### Station integrity principles
+
+The [OpenTelemetry library guidelines (point
+4)](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md#requirements)
+describes a separation of protocol-dependent ("receivers",
+"exporters") and protocol-independent ("processors") parts.  We refer
+to the combination of parts as a station.
+
+The station concept is called out because within a station, we expect
+that the station (software) acts responsibly by design, for the
+integrity of the pipeline.  Stations allow data to enter a pipeline
+only through receiver components.  Stations are never responsible for
+dropping data, because only processor components drop data.  Stations
+allow data to leave a pipeline only through exporter components.
+
+Because of station integrity, we can make the following assertions:
+
+1. Data that enters a pipeline is eventually exported or dropped.
+2. No other outcomes are possible.
+
+There is a potential for pipeline metrics to be redundant, as
+described in these assertions.  In a pipeline with no fan-in or
+fan-out, each stage processes as many items as the stage before it
+did, minus the number of items dropped.
+
+#### Pipeline stage-name uniqueness
+
+The Pipeline Stage Name Uniqueness requirement developed here avoids
```

Comment:
This requirement doesn't seem necessary. A single metric name often counts the same thing from the perspective of multiple entities (e.g. faas.invocations will overcount if you have faas instances that talk to other faas instances).  Isn't resource enough to distinguish multiple chained sdk + collector(s)?

Reply:
@dashpole thanks for an early review -- it was still a pretty rough draft that you saw but I've just updated it, and the detail you flagged here doesn't change. 

I am specifically concerned about wanting to automatically infer the health of the pipeline from all the pipeline-monitoring metrics. I'm worried about the 3-level layout that could be standard, with sdk sending to agent sending to gateway collectors.  If the agent and the gateway count items using the same metric name and I enter a metrics system with a query like `rate(otel.collector.received)` and compare it against `rate(otel.sdk.received)`, the number of collector items will be exactly double the number of SDK items. The resources would distinguish which was which, but not in a way that I can automatically apply to avoid this "incorrect" interpretation. I'm following the rule that it should be difficult to form meaningless aggregates. 

With that said, I don't want a dashboard problem and I don't expect more than 3 sets of 3 metric names, i.e., {sdk,agent,gateway} x {received,dropped,exported}. If a user were to configure a custom station name, they would just have to create a custom dashboard.